### PR TITLE
refactor: expose configDef() for all cache config sets

### PR DESCRIPTION
**Pull request by jeqo**
This is needed for documentation purposes (#601), while also improving the cache config builder.

### CODE REVIEW

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java:1]: I'd start review here: this is the main change affecting the cache config hierarchy. 
Then continue with MemorySegmentIndexes/MemorySegmentManifest configs, and then ChunkCacheConfig and its children

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java [hunk 1]
```diff
@@ -38,59 +38,21 @@ public class CacheConfig extends AbstractConfig {
...
-    static final long CACHE_RETENTION_MS_DEFAULT = 600_000;
+    public static final long CACHE_RETENTION_MS_DEFAULT = 600_000;
 
-    private static ConfigDef configDef(
+    public CacheConfig(
         final ConfigDef configDef,
-        final Object defaultSize,
-        final long defaultRetentionMs
+        final Map<String, ?> props
     ) {
```
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java:46]: Disentangle configDef from CacheConfig. Instead the `DefBuilder` defines the config definition.

File: core/src/test/java/io/aiven/kafka/tieredstorage/config/CacheConfigTest.java
Comment (jeqo) [inline core/src/test/java/io/aiven/kafka/tieredstorage/config/CacheConfigTest.java:1]: Just adjusting new instantiation.

**Review by biggusdonzus — APPROVED** [review]
Code looks good, but i don't understand some of the changes like why in `CacheConfig.java` why move the `configDef.define` defs into the `DefBuilder` class? But maybe that's necessary for the documentation feature, so in the end it's just my ramblings :smiley:

**Pull request merged**